### ISSUE TITLE
Variable font not working

### ISSUE THREAD

**Issue by JDPlantation** [tokotype/PlusJakartaSans#15]
After installing the variable font file, only extra light is installed. Or is that how it's supposed to be and I'm being stupid?

**Comment by kenmcd** [comment]
Does the application you are using support variable fonts?
When an application does not support variable fonts what is displayed is the default master.
And in this variable font the default master is set to ExtraLight (usually this is the Regular).

**Comment by JDPlantation** [comment]
This was in OBS, where other fonts have variable styles
Also, when installing, the install window that appears to preview the font, all the preview pages are in Extra Light. 
I appreciate the help :)

**Comment by kenmcd** [comment]
> Also, when installing, the install window that appears to preview the font, all the preview pages are in Extra Light.

Oh, I see what you mean. I see the same thing when looking in the Windows Font Viewer.
I tried changing the default to Regular, and added a SemiBold instance, and checked the names.
Exported again. Same thing.
I can see in the Viewer it will cycle through the correct number of fonts, but they are all ExtraLight.
Very weird.
Have to play with it some more tomorrow.

**Comment by JDPlantation** [comment]
Ah ok. So it's not a case of me being an idiot!

**Comment by kenmcd** [comment]
Cannot figure-out how to fix this.
I renamed some things and added a SemiBold.
Exporting the static fonts works just fine.
(appears to work fine, but based on what I have seen the fonts should be tested extensively).

I got the variable to show three weights, Light, Regular, and Bold (which are the masters).
But it still cycles thru all the instances showing just those three weights.
I think is is because of a bunch of layers in the masters, which may require the Glyphs app to fix.
Using FontLab I tried to make some changes to the layers, but my changes caused certain glyphs to disappear.

So your only hope is if the original designers fix the variable font.

**Comment by JDPlantation** [comment]
ok! Thanks anyway. Is there a way for me to grab the variable with light regular and bold? That will be more than enough for me :)

**Comment by kenmcd** [comment]
> ok! Thanks anyway. Is there a way for me to grab the variable with light regular and bold? That will be more than enough for me :)

I figured-out the cause of the problems.
Now I am figuring-out how to fix it.
Hopefully I will have it fixed today or tomorrow.
Will post it here when done.

I really do not want to post the partially fixed version.
So ya gotta wait. :-)

**Comment by JDPlantation** [comment]
Aha! You're the man! No rush, thank you so much for taking the time!

**Comment by kenmcd** [comment]
@JDPlantation 
OK. Try these. This is just the two variable fonts.
[Plus Jakarta Sans LO.zip](https://github.com/tokotype/PlusJakartaSans/files/6475786/Plus.Jakarta.Sans.LO.zip) (ZIP, 157KB)
You are now appointed official tester - because I am tried of looking at it. ;-)
They do work properly in the Windows Font Viewer now.
Remember the Regular and the Bold are in the same style group so they both say just _Plus Jakarta Sans LO._
Let me know how it works in OBS, and anywhere else you want to test.

Note: the LO is for LibreOffice; I add that to the name when I fix static fonts for LO.
I also added SemiBold to the variable and the statics.

**Comment by JDPlantation** [comment]
You, good sir, are as kind as you are wise!

They all install ok, completely usable. In windows font settings, they all appear as intended.

In OBS they appear as separate fonts (Except for regular and bold) but I think it's supposed to do that anyway. (Also checked this in Word and Ilustrator, and it's the same)
![image](https://user-images.githubusercontent.com/84051106/118209440-fec18700-b43e-11eb-8182-358e75219ebd.png)

**Comment by kenmcd** [comment]
Cool! 
I do not see the italics in the OBS (?) screenshot.
Did you install the Italic font too?

**Comment by JDPlantation** [comment]
I hadn't installed it yet! 
After installing the italics font, in the Windows Font viewer it's correct.
![image](https://user-images.githubusercontent.com/84051106/118403705-04d58480-b646-11eb-804d-e176dda39b65.png)
However, in OBS it only appears as on option in regular, and doesn't work
![image](https://user-images.githubusercontent.com/84051106/118403450-c12e4b00-b644-11eb-8f9d-a3cbf67aadb7.png)
I also checked in Adobe Illustrator, and the italics option doesn't even appear.

I'm personally not too worried about having italics, as I won't use it in OBS, but if you're anything like me, and not knowing why it doesn't work bugs you, I'm more than happy to be your tester lol

(I'm completely clueless as to the inner workings of this stuff so I don't know if this makes a difference, but I noticed in the windows font installer preview window, the font name is the same for the italics version and the normal version. Could that cause a conflict when programs are trying to find the font?)

![image](https://user-images.githubusercontent.com/84051106/118403829-ac52b700-b646-11eb-8272-9ccc0a2dbd49.png)

**Comment by kenmcd** [comment]
> the font name is the same for the italics version and the normal version. Could that cause a conflict when programs are trying to find the font?)

Yes, that is how it is supposed to work in Windows apps which use style groups (like Word, LibreOffice).
That is why I wanted to see how the italics was working.
The Regular and the Italic are in the two separate files, but they should work together.

The OBS issue may be just an OBS issue, or I still need to do more to fix the fonts.
I can test it in Word later today and see if that is working (which means I got the stat table correct)..
Apps like IL use different tables inside the variable fonts so that may also need more work.

Thanks for the tests and the images. 
At least the Regular is now working in OBS!
And since I am now 100%-Adobe-Free it helps that you can test in IL.

So I will keep going on this; may take a few days. :-)

**Comment by JDPlantation** [comment]
I tested in wordpad, but I the italics worked in wordpad even when the italic font wasn't installed, so I don't know if wordpad has some weird way of making anything italic lol
Out of curiosity, what do you use instead of Adobe? I have illustrator for lack of knowing other software (I'm not a designer, I use it now and again to make images for my stream) but I don't find it very user friendly

**Comment by kenmcd** [comment]
> I tested in wordpad, but I the italics worked in wordpad even when the italic font wasn't installed, so I don't know if wordpad has some weird way of making anything italic lol

Yeah, need to be cautious when testing of the apps that do fake italics.
Thanks for reminding me.
Usually I export to PDF and check the embedded fonts there to see if anything was faked.
Be interesting to test embedding this font, because embedding variable fonts is also kinda "variable".

> what do you use instead of Adobe?

After using them for many years I really do not like the whole Adobe thing anymore.
Currently 100%-Adobe-Free on this new laptop - never installed any of it at all on here.

Been closely following the development of the Affinity apps.
But they do not yet support variable fonts.
In a discussion in the Affinity forum today I realized I do really need to have InDesign for testing things.
So I will install it and IL in a VM to keep my system clean of all Adobe hidden crap.

Installed Corel the other day for font testing because it does also support variable fonts.
But I have not even touched it again since installing it.

Downloaded a beta yesterday of VectorStyler to test because it has variable font support.
Have not even installed it yet.

Part of the problem of testing variable fonts is all the apps have their issues too in their VF support.
VF support is still very much a moving target.

So to get back to your original question, my plan long-term is to use the Affinity apps when all the features I need are available.
